readiness-tool/popup.js

```diff
@@ -0,0 +1,261 @@
+      'analytics': [],
+      'other': [],
+    },
+  };
+
+  // for all the app objects in the apps.JSON file
+  Object.keys(obj).forEach(function(key) {
+    let val = obj[key];
+    // If object has a 'script' key
+    if (val.script != null) {
+      // Sometimes val.script contains an array of regular expressions
+      if (typeof val.script == 'object') {
+        Object.keys(val.script).forEach(function(x) {
+          addToDict(val.script[x], htmlString, foundThis, key, val.cats);
+        });
+      } else {
+        addToDict(val.script, htmlString, foundThis, key, val.cats);
+      }
+    }
+  });
+
+  return foundThis;
+}
+
+
+/**
+ * Pushes keys to the supported or not supported list of the object 'found this'
+ * @param {String} tempScript - String representation of regular expression
+ * @param {!String} htmlString - String containing all HTML on the page
+ * @param {Object} foundThis - Object separating the 3P services by support
+ * @param {String} key - name of third party service
+ * @param {String} category - the category that the key belongs to
+ */
+function addToDict(tempScript, htmlString, foundThis, key, category) {
+  tempScript = tempScript.split('\\;');
+  let regX = new RegExp(tempScript[0]);
+  if (doesRegexExist(regX, htmlString)) {
+    if (isKeyUnique(foundThis, key)) {
+      if (isSupported(key)) {
+        findCategory(category, foundThis.supported, key);
+      } else {
+        findCategory(category, foundThis.notSupported, key);
+      }
+    }
+  }
+}
+
+/**
+ * Checks to see if key is unique within the object
+ * @param {Object} obj - Object separating the 3p services by support
+ * @param {String} key - name of third party service
+ * @return {boolean} 
+ */
+function isKeyUnique(obj, key) {
+  let truthValue = obj.supported.ads.indexOf(key) +
+      obj.supported.analytics.indexOf(key) + obj.notSupported.ads.indexOf(key)+
+      obj.notSupported.analytics.indexOf(key);
+  return truthValue == -4;
+}
+
+/**
+ * Figures out thae category the key belongs to (Ads or Analytics)
+ * @param {String} category - number representing the category of each item
+ * @param {Object} objectList - The found supported/unsupported 3p services
+ * @param {String} key - name of third party service
+ */
+function findCategory(category, objectList, key) {
+  // '10' is Analytics
+  if (category == '10') {
+    objectList.analytics.push(key);
+    // '36' is Ads
+  } else if (category == '36') {
+    objectList.ads.push(key);
+  } else {
+    objectList.other.push(key);
+  }
+}
+
+
+/**
+ * Determines if regular expression exists within the HTML of the page
+ * @param {String} regexString - Regular expression of app
+ * @param {String} htmlString - String containing all HTML on the page
+ * @return {boolean} 
+ */
+function doesRegexExist(regexString, htmlString) {
+  let value = regexString.test(htmlString);
+  return (value);
+}
+
+/**
+ * TODO (alwalton@): get list of supported ads/analytics programatically
```

Comment:
OOC, how are you planning to do this? Make sure you specify this in your design doc.